src/components/ChartWidgetFilter/ChartWidgetFilter.tsx

```diff
@@ -25,29 +29,33 @@ const ChartWidgetFilter: FC<Props> = ({ widgetId }) => {
 
   const datePickerData = useSelector((state: RootState) => {
     if ('datePickerId' in widget && widget.datePickerId) {
-      const { data, isActive } = getWidget(state, widget.datePickerId);
+      const { data, isActive, widget: datePickerWidget } = getWidget(
+        state,
+        widget.datePickerId
+      );
       if (isActive) {
-        return data;
+        const filterWidget = datePickerWidget as DatePickerWidget;
+        return { data, name: filterWidget.settings.name };
       }
-
       return null;
     }
-
     return null;
   });
 
   const filtersData = useSelector((state: RootState) => {
     if ('filterIds' in widget && widget.filterIds.length > 0) {
       const filters = widget.filterIds.reduce(
         (activeFilters: FilterMeta[], id: string) => {
-          const { data, isActive } = getWidget(state, id);
+          const { data, isActive, widget } = getWidget(state, id);
+          const filterWidget = widget as FilterWidget;
```

Comment:
I suggest to move:
```
const filterWidget = widget as FilterWidget;
```
To the condition body and add additional condition case: `type === 'filter'`. This will allow TypeScript to automatically infer the FilterWidget and remove manual casting. 

The same pattern could be used for date picker widget.